### PR TITLE
Remove legacy renderer dependencies in LyShine

### PR DESCRIPTION
**Pull request by michabr**
Remove legacy IRenderer.h dependencies
Remove legacy VertexFormat.h dependencies

Notes
- UiTextComponent still references the legacy struct SVF_P2F_C4B_T2F_F4B because it's currently still being used in AtomFont
- A new file was added to CryCommon. This is temporary until all Lyshine interfaces are moved to the gem

Signed-off-by: abrmich <abrmich@amazon.com>

### CODE REVIEW

**Review by bosnichd — APPROVED** [review]
Great, thanks!

File: Code/Legacy/CryCommon/LyShine/IRenderGraph.h [hunk 1]
```diff
@@ -7,9 +7,9 @@
  */
 #pragma once
 
-#include <IRenderer.h>
 #include <ITexture.h>
```
Comment (VickyAtAZ) [inline Code/Legacy/CryCommon/LyShine/IRenderGraph.h:10]: Can you remove dependency of ITexture.h as well? I think that's also part of cry renderer.

File: Gems/LyShine/Code/Editor/SpriteBorderEditorCommon.h [hunk 2]
```diff
@@ -10,7 +10,6 @@
 #include <vector> // required to be included before platform.h
 #include <platform.h>
 #include <ISystem.h>
```
Comment (VickyAtAZ) [inline Gems/LyShine/Code/Editor/SpriteBorderEditorCommon.h:12]: Do you have plan to remove the dependency of CrySystem? 
I think AZCore should provide everything you may need to replace them.

File: Code/Legacy/CryCommon/LyShine/IRenderGraph.h (hunk 1 above)
Comment (michabr) [reply Code/Legacy/CryCommon/LyShine/IRenderGraph.h]: Missed this one, thanks! I've removed it.

File: Gems/LyShine/Code/Editor/SpriteBorderEditorCommon.h (hunk 2 above)
Comment (michabr) [reply Gems/LyShine/Code/Editor/SpriteBorderEditorCommon.h:12]: Yes, CrySystem dependency will be addressed separately.

**Review by VickyAtAZ — APPROVED** [review]
Looks good overall. 

I would suggest adding LyShine namespace for all the classes in the gem eventually.

**Review by cgalvan — APPROVED** [review]

**Pull request closed without merging**